tools/cellprofiler/runcp.py

```diff
@@ -0,0 +1,40 @@
+import subprocess
+import sys
+import os
+from random import randint
+
+DEVNULL = open(os.devnull,'wb') #This is needed to supress ilastik error due to running CellProfiler in headless configuration
+cp_path = os.environ['CP_PATH']
+outpath = str(randint(1,100000))
+tmppath = cp_path + '/bcc/' + outpath 
+if(sys.argv > 3):
+	print "Too many command line arguments"
+if(sys.argv[1] == 'bcc'):
+	subprocess.call('mkdir '+tmppath,shell=True)
+	subprocess.call('mkdir '+tmppath+'/output',shell=True)
+	os.chdir(tmppath)
```

Comment:
Remember that the path $CP_PATH/tmp may not have write permissions for the user in general.
When Galaxy runs a job, it creates a unique directory for every job (something like /cluster_share/Galaxy/database/job_working_directory/000/<uniq_id>/). When this python script runs, its initial working directory is the unique directory created by Galaxy. Also, it is guaranteed that the user will have write permissions to this directory.
Will these scripts work if the working directory is the one created by Galaxy and you run them with a full path?
Example: python CP_PATH/createSymLinks.py